crates/red_knot/src/lint.rs

```diff
@@ -150,17 +149,16 @@ fn lint_bad_override(context: &SemanticLintContext, class: &ast::StmtClassDef) {
             return;
         };
 
-        if ty.has_decorator(&typing_context, override_ty) {
-            let method_name = ty.name(&typing_context);
-            if class_ty
-                .inherited_class_member(&typing_context, method_name)
-                .is_none()
-            {
+        let db = semantic.db();
```

Comment:
Okay, This is a bit problematic because I don't want to give lint rules access to the `db`, because I don't consider the `db` to be part of the stable API. 

The solution for `has_decorator` is relatively simple, we make it a method on `model`: `model.has_decorator(ty, decorator_ty)`. It's less clear to me me how we remove the `db` dependency from `name`, and `inherited_class_member`. Can we change `has_decorator` and add a TODO for the other methods.

Reply:
After offline discussion, I'm going to just add a TODO here; we need to explore this further and find a general solution, and it's not clear that we'd even want `has_decorator` to be a method directly on the `SemanticModel` once we have that general solution.